crowdstrike/table_crowdstrike_detect.go

```diff
@@ -0,0 +1,222 @@
+package crowdstrike
+
+import (
+	"context"
+
+	"github.com/crowdstrike/gofalcon/falcon"
+	"github.com/crowdstrike/gofalcon/falcon/client"
+	"github.com/crowdstrike/gofalcon/falcon/client/detects"
+	"github.com/crowdstrike/gofalcon/falcon/models"
+	"github.com/turbot/steampipe-plugin-sdk/v3/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v3/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v3/plugin/transform"
+	"golang.org/x/time/rate"
+)
+
+//// TABLE DEFINITION
+
+func tableCrowdStrikeDetects(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "crowdstrike_detect",
+		Description: "Detections are events identified by Falcon sensors on the hosts in your environment.",
+		List: &plugin.ListConfig{
+			Hydrate: listCrowdStrikeDetects,
+			KeyColumns: []*plugin.KeyColumn{
+				{
+					Name:      "created_timestamp",
+					Require:   plugin.Optional,
+					Operators: []string{">", ">=", "=", "<", "<="},
+				},
+				{
+					Name:      "first_behavior",
+					Require:   plugin.Optional,
+					Operators: []string{">", ">=", "=", "<", "<="},
+				},
+				{
+					Name:      "last_behavior",
+					Require:   plugin.Optional,
+					Operators: []string{">", ">=", "=", "<", "<="},
+				},
+				{
+					Name:      "status",
+					Require:   plugin.Optional,
+					Operators: []string{"="},
+				},
+				{
+					Name:      "detection_id",
+					Require:   plugin.Optional,
+					Operators: []string{"="},
+				},
+				{
+					Name:      "max_severity",
+					Require:   plugin.Optional,
+					Operators: []string{">", ">=", "=", "<", "<="},
+				},
+				{
+					Name:      "max_confidence",
+					Require:   plugin.Optional,
+					Operators: []string{">", ">=", "=", "<", "<="},
+				},
+			},
+		},
+		Columns: []*plugin.Column{
+			{Name: "adversary_ids", Description: "If behaviors or indicators in a detection are attributed to an adversary that is tracked by CrowdStrike Falcon Intelligence, those adversaries will have an ID associated with them. These IDs are found in a detection's metadata which can be viewed using the Detection Details API.", Type: proto.ColumnType_JSON, Transform: transform.FromJSONTag()},
+			{Name: "assigned_to_name", Description: "The human-readable name of the user to whom the detection is currently assigned.", Type: proto.ColumnType_STRING},
+
+			{Name: "assigned_to_uid", Description: "TODO", Type: proto.ColumnType_STRING, Transform: transform.FromJSONTag()},
+			{Name: "behaviors", Description: "TODO", Type: proto.ColumnType_JSON},
+			{Name: "behaviors_processed", Description: "TODO", Type: proto.ColumnType_JSON},
+
+			{Name: "cid", Description: "Your organization's customer ID (CID).", Type: proto.ColumnType_STRING},
+
+			{Name: "created_timestamp", Description: "TODO", Type: proto.ColumnType_TIMESTAMP, Transform: transform.From(func(ctx context.Context, td *transform.TransformData) (interface{}, error) {
```

Comment:
Can we move this transform function into `utils.go`? It makes the column definitions difficult to read